src/silx/gui/dialog/AbstractDataFileDialog.py

```diff
@@ -443,7 +443,7 @@ def saveState(self):
         stream.writeInt32(self.__serialVersion)
         stream.writeQVariant(self.__detailView.header().saveState())
         viewMode = self.viewMode()
-        if qt.BINDING == 'PyQt6':  # No auto conversion to int
+        if qt.BINDING in ('PyQt6', 'PySide6'):  # No auto conversion to int
```

Comment:
@t20100 this probably comes with this Enum redefinition. So this mean this code will only work for recent PySide6 version. But couldn't find the version

Reply:
error is:
```
        viewMode = self.viewMode()
        if qt.BINDING == 'PyQt6':  # No auto conversion to int
            viewMode = viewMode.value
>       stream.writeInt32(viewMode)
E       TypeError: 'PySide6.QtCore.QDataStream.writeInt32' called with wrong argument types:
E         PySide6.QtCore.QDataStream.writeInt32(ViewMode)
E       Supported signatures:
E         PySide6.QtCore.QDataStream.writeInt32(int)

```

Reply:
looks like matplotlib get this kind of handling:
```
    elif QT_API == QT_API_PYSIDE6:
        from PySide6 import QtCore, QtGui, QtWidgets, __version__
        import shiboken6
        def _isdeleted(obj): return not shiboken6.isValid(obj)
        if parse_version(__version__) >= parse_version('6.4'):
            _to_int = operator.attrgetter('value')
        else:
            _to_int = int
```